src/components/SearchModalBridge/CurrencySearch.tsx

```diff
@@ -0,0 +1,224 @@
+
+  const sortedTokens: Token[] = useMemo(() => {
+    return filteredTokens.sort(tokenComparator)
+  }, [filteredTokens, tokenComparator])
+
+  const filteredSortedTokens = useSortedTokensByQuery(sortedTokens, debouncedQuery)
+
+  const filteredSortedTokensWithNativeCurrency: Currency[] = useMemo(() => {
+    if (!showNativeCurrency) return filteredSortedTokens
+    const s = debouncedQuery.toLowerCase().trim()
+    if (nativeCurrency.symbol && nativeCurrency.symbol.toLowerCase().startsWith(s)) {
+      return nativeCurrency ? [nativeCurrency, ...filteredSortedTokens] : filteredSortedTokens
+    }
+    return filteredSortedTokens
+  }, [showNativeCurrency, filteredSortedTokens, debouncedQuery, nativeCurrency])
+
+  const handleCurrencySelect = useCallback(
+    (currency: Currency) => {
+      onCurrencySelect(currency)
+      onDismiss()
+    },
+    [onDismiss, onCurrencySelect]
+  )
+
+  // clear the input on open
+  useEffect(() => {
+    if (isOpen) setSearchQuery('')
+  }, [isOpen])
+
+  // manage focus on modal show
+  const inputRef = useRef<HTMLInputElement>()
+  const handleInput = useCallback(event => {
+    const input = event.target.value
+    const checksummedInput = isAddress(input)
+    setSearchQuery(checksummedInput || input)
+    fixedList.current?.scrollTo(0)
+  }, [])
+
+  const handleEnter = useCallback(
+    (e: KeyboardEvent<HTMLInputElement>) => {
+      if (e.key === 'Enter') {
+        const s = debouncedQuery.toLowerCase().trim()
+        if (s === nativeCurrency.symbol) {
+          handleCurrencySelect(nativeCurrency)
+        } else if (filteredSortedTokensWithNativeCurrency.length > 0) {
+          if (
+            filteredSortedTokensWithNativeCurrency[0].symbol?.toLowerCase() === debouncedQuery.trim().toLowerCase() ||
+            filteredSortedTokensWithNativeCurrency.length === 1
+          ) {
+            handleCurrencySelect(filteredSortedTokensWithNativeCurrency[0])
+          }
+        }
+      }
+    },
+    [debouncedQuery, filteredSortedTokensWithNativeCurrency, handleCurrencySelect, nativeCurrency]
+  )
+
+  // menu ui
+  const [open, toggle] = useToggle(false)
+  const node = useRef<HTMLDivElement>()
+  useOnClickOutside(node, open ? toggle : undefined)
+
+  // if no results on main list, show option to expand into inactive
+  const filteredInactiveTokens = useSearchInactiveTokenLists(
+    filteredTokens.length === 0 || (debouncedQuery.length > 2 && !isAddressSearch) ? debouncedQuery : undefined
+  )
+  const filteredInactiveTokensWithFallback = useMemo(() => {
+    if (filteredTokens.length > 0) return []
+    if (filteredInactiveTokens.length > 0) return filteredInactiveTokens
+    if (searchToken) return [searchToken]
+    return []
+  }, [filteredInactiveTokens, filteredTokens.length, searchToken])
+
+  useEffect(() => {
+    inputRef.current?.focus()
+  }, [inputRef])
+
+  return (
+    <ContentWrapper>
+      <AutoColumn style={{ padding: '22px 18.5px 20px 18.5px' }} gap="15px">
```

Comment:
Also same as last comment lets try to reuse components these seem almost exactly the same, would be good to dry up and reuse as much as we can..
![Screenshot 2022-04-06 at 15 37 37](https://user-images.githubusercontent.com/33226956/161987812-2cbe8d88-d87c-4b2e-ad44-fc2f55705a7f.png)

Reply:
Fixed in #908